sourcecode/src-dotty/sourcecode/Macros.scala

```diff
@@ -109,9 +109,9 @@ object Macros {
     val owner = actualOwner(qctx)(rootContext.owner)
     val fullName =
       owner.fullName.trim
-        .split("\\.", -1)
+        .split("\\.", -1).map(_.nn)
         .filterNot(Util.isSyntheticName)
-        .map(_.split("_\\$", -1).dropWhile(_.isEmpty).mkString.stripSuffix("$")) // meh
+        .map(_.split("_\\$", -1).map(_.nn).dropWhile(_.isEmpty).mkString.stripSuffix("$")) // meh
```

Comment:
**Java method returns array**
Java method returns array of type `Array[Null | T]`. We need to map it to `Array[T]`.